### PR TITLE
DS-3829: Add array indices to QStatisticsTestingInfo attr

### PR DESCRIPTION
**Pull request by mwmclean**

### CODE REVIEW

File: R/subscript-tables.R [hunk 1]
```diff
@@ -268,6 +271,62 @@ getQTestInfoIndexForVectorOutput <- function(evaluated.args, dimnames.x, perm)
...
+                          logical(1L))
+        if (all(dropped))  # y reduced to a single element, drop indices in attr.
+            return(q.test.info[, !col.idx, drop = FALSE])
+        arr.idx <- arr.idx[, !dropped, drop = FALSE]
+
+        ## Update dimension names preserving column order
+        col.ord <- order(match(colnames(arr.idx), qtable.dim.names.allowed)[!dropped])
+        colnames(arr.idx) <- qTableDimensionNames(dim.len)[col.ord]
+    }else
+        return(q.test.info)
```
[the remark this answers is not in the document]
Comment (jrwishart) [reply R/subscript-tables.R:312]: Also, can check early if `y` is a single element and then `return(q.test.info[, !col.idx, drop = FALSE])` without the need to check all are unique single values in each column. Perhaps before the `dim.len < length(dim.names)` block.

File: tests/testthat/test-subscript-tables.R [hunk 2]
```diff
@@ -856,5 +857,70 @@ test_that("DS-3837: Can subset QTables missing dimnames",
...
+
+    ## Single column from a 2D crosstab, drop = FALSE
+    expected.index <- expand.grid(dimnames.tbl[row.major.idx])
+    q.test.info <- attr(tbl, "QStatisticsTestingInfo")
+    expected <- cbind(expected.index, q.test.info)
+    keep.idx <- nrow(tbl)*(0:(ncol(tbl)-1)) + 1
+    expected <- expected[keep.idx, ]
+    out <- tbl[, 1, drop = FALSE]
+    q.test.info.out <- attr(out, "QStatisticsTestingInfo")
+    expect_equal(q.test.info.out, expected)
```
Comment (jrwishart) [inline tests/testthat/test-subscript-tables.R:913]: The names of the expected attribute here are in the order of `c("Column", "Row")` should this be reversed to have row first?

**Review by jrwishart — CHANGES_REQUESTED** [review]

**Comment by jrwishart** [comment]
As a general aside. This looks good for a dev to validate the data structure in `R` but the lookup is never actually used in subscripting. That is, it seems a good code validation tool. Perhaps keep this code but only use it for debug purposes?

**Comment by mwmclean** [comment]
Just because it's not used now, doesn't mean it won't be useful later. If anyone at all ever inspects this `QStatisticsTestingInfo` attribute, the extra info is clearly helpful to understand how what is stored in the attribute matches up to the table. Really the Core team should have given us this info from the start in C#. I'd prefer to leave it in.

**Comment by jrwishart** [comment]
> Just because it's not used now, doesn't mean it won't be useful later. If anyone at all ever inspects this `QStatisticsTestingInfo` attribute, the extra info is clearly helpful to understand how what is stored in the attribute matches up to the table. Really the Core team should have given us this info from the start in C#. I'd prefer to leave it in.

I want to keep it too but was thinking not have it executed each time. Hidden behind a debug flag perhaps?

File: R/subscript-tables.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (mwmclean) [reply R/subscript-tables.R]: Implemented these two suggestions in c1545d2

File: tests/testthat/test-subscript-tables.R (hunk 2 above)
Comment (mwmclean) [reply tests/testthat/test-subscript-tables.R]: I modified the column order of the indices in the output to rectify this in c1545d2.

Comment (mwmclean) [reply tests/testthat/test-subscript-tables.R]: Actually fixed in https://github.com/Displayr/verbs/commit/9a936fde4972a1cd3dfa5c9d7d41c75472086ff3

**Comment by mwmclean** [comment]
I don't see the benefit of having them off. The storage and computational costs are minimal to none.

> > Just because it's not used now, doesn't mean it won't be useful later. If anyone at all ever inspects this `QStatisticsTestingInfo` attribute, the extra info is clearly helpful to understand how what is stored in the attribute matches up to the table. Really the Core team should have given us this info from the start in C#. I'd prefer to leave it in.
> 
> I want to keep it too but was thinking not have it executed each time. Hidden behind a debug flag perhaps?

I added a logical argument to `[.qTable` for you in 75de08c, but I set it to add the indices by default, because there's basically no storage or computational overhead to adding them.

**Review by jrwishart — APPROVED** [review]

**Pull request merged**